src/plots/cartesian/axes.js

```diff
@@ -2636,6 +2638,35 @@ axes.drawOne = function(gd, ax, opts) {
     return Lib.syncOrAsync(seq);
 };
 
+function keepSelectedAutoMargin(automargin, push) {
+    if(typeof automargin === 'boolean') return push;
+
+    var keepMargin = [];
+    var mapping = {
+        width: ['x', 'r', 'l'],
+        height: ['y', 't', 'b'],
+        right: ['r'],
+        left: ['l'],
+        top: ['t'],
+        bottom: ['b']
+    };
+
+    Object.keys(mapping).forEach(function(key) {
+        if(automargin.includes(key)) {
+            mapping[key].forEach(function(item) {
+                keepMargin.push(item);
+            });
+        }
+    });
+
+    Object.keys(push).forEach(function(key) {
+        if(key.length !== 2 && !keepMargin.includes(key)) {
```

Comment:
What's the `length !== 2` about?

Reply:
OIC there's `xl`, `xr`, `yt`, `yb`. But don't we want to always include `x` and `y` as well? I've forgotten exactly what those do, and doesn't look like we ever documented this... but it does appear that `x` and `y` are the fallbacks when for whatever reason we didn't provide the more specific 2-letter variants

https://github.com/plotly/plotly.js/blob/b194d8c3ef503ac1989666547cccb65dc60fb44f/src/plots/plots.js#L1931-L1934

Reply:
Yes, it is for these properties. Corrected the code, made the removal of the property if it  is not in the keep list.